storage/src/journal/fixed.rs

```diff
@@ -1449,4 +1535,385 @@ mod tests {
+                assert_eq!(synced_journal.oldest_retained_pos().await.unwrap(), Some(0));
+                // no blobs created for pruned range, partial data in tail at index 0
+                assert_eq!(synced_journal.blobs.len(), 0);
+                assert_eq!(synced_journal.tail_index, 0);
+                assert_eq!(
+                    synced_journal.tail.size().await,
+                    PRUNED_SIZE * Journal::<Context, Digest>::CHUNK_SIZE_U64
+                );
+
+                synced_journal.destroy().await.unwrap();
+            }
+
+            // Test Case 3: Initialize with pruned state exactly at blob boundary
+            {
+                let cfg = Config {
+                    partition: "test_pruned_boundary".into(),
+                    items_per_blob: ITEMS_PER_BLOB,
+                    write_buffer: WRITE_BUFFER,
+                    buffer_pool: PoolRef::new(PAGE_SIZE, PAGE_CACHE_SIZE),
+                };
+
+                let synced_journal = Journal::<Context, Digest>::init_pruned(
+                    context.clone(),
+                    cfg.clone(),
+                    ITEMS_PER_BLOB, // Exactly one full blob
+                )
+                .await
+                .expect("Failed to init with pruned state at blob boundary");
+
+                // Verify the synced journal has correct state
+                assert_eq!(synced_journal.size().await.unwrap(), ITEMS_PER_BLOB);
+                assert_eq!(synced_journal.oldest_retained_pos().await.unwrap(), None);
+                // Boundary case: no blobs created for pruned range, empty tail at next index
+                assert_eq!(synced_journal.blobs.len(), 0);
+                assert_eq!(synced_journal.tail_index, 1);
+                assert_eq!(synced_journal.tail.size().await, 0);
+
+                synced_journal.destroy().await.unwrap();
+            }
+
+            // Test Case 4: Initialize with pruned state spanning multiple blobs
+            {
+                const MULTI_BLOB_SIZE: u64 = 14; // 2 full blobs + 4 items in third blob
+
+                let cfg = Config {
+                    partition: "test_pruned_multi".into(),
+                    items_per_blob: ITEMS_PER_BLOB,
+                    write_buffer: WRITE_BUFFER,
+                    buffer_pool: PoolRef::new(PAGE_SIZE, PAGE_CACHE_SIZE),
+                };
+
+                let synced_journal = Journal::<Context, Digest>::init_pruned(
+                    context.clone(),
+                    cfg.clone(),
+                    MULTI_BLOB_SIZE,
+                )
+                .await
+                .expect("Failed to init with pruned state spanning multiple blobs");
+
+                // Verify the synced journal state
+                assert_eq!(synced_journal.size().await.unwrap(), MULTI_BLOB_SIZE);
+                assert_eq!(
+                    synced_journal.oldest_retained_pos().await.unwrap(),
+                    Some(10)
```

Comment:
Just want to flag this behavior -- we report that the oldest_retained_pos() is `Some(10)` even though it's actually dummy data. We tell the caller of `init_pruned`: 
```
 /// - Tail blob is filled with dummy data up to its size -- this shouldn't be read.
```

I don't think this is an issue for how we use the fixed journal in syncing.